aiocache/factory.py

```diff
@@ -79,25 +88,98 @@ def create(self, alias=None, cache=None, **kwargs):
-                "caches.set_config specifying the config for that cache".format(alias))
+                "Could not find config for '{0}', ensure you include {0} "
+                "when calling caches.config setter specifying the config "
+                "for that cache".format(alias)
+            )
+
+        return MappingProxyType(self.config[alias])
+
+    @property
+    def config(self) -> MappingProxyType:
+        """
+        Current stored config.
+
+        :getter: Return mapping proxy to config
+        :setter:
+            Set (override) the default config for cache aliases
+            from a dict-like structure. The structure is the following::
+
+                {
+                    'default': {
+                        'cache': "aiocache.SimpleMemoryCache",
+                        'serializer': {
+                            'class': "aiocache.serializers.StringSerializer"
+                        }
+                    },
+                    'redis_alt': {
+                        'cache': "aiocache.RedisCache",
+                        'endpoint': "127.0.0.10",
+                        'port': 6378,
+                        'serializer': {
+                            'class': "aiocache.serializers.PickleSerializer"
+                        },
+                        'plugins': [
+                            {'class': "aiocache.plugins.HitMissRatioPlugin"},
+                            {'class': "aiocache.plugins.TimingPlugin"}
+                        ]
+                    }
+                }
 
-        return config[alias]
+            'default' key must always exist when passing a new config.
+            Default configuration is::
+
+                {
+                    'default': {
+                        'cache': "aiocache.SimpleMemoryCache",
+                        'serializer': {
+                            'class': "aiocache.serializers.StringSerializer"
+                        }
+                    }
+                }
+
+            You can set your own classes there.
+            The class params accept both str and class types.
+
+            All keys in the config are optional, if they are not passed the defaults
+            for the specified class will be used.
+
+            If a config key already exists, it will be updated with the new values.
+        :type: MappingProxyType
+        """
+        return MappingProxyType(self._config)
 
     def get_config(self):
         """
-        Return copy of current stored config
+        .. deprecated:: 0.9.1
+            Use ``config`` property getter instead.
+
+        Return a proxy to current stored config
         """
-        return deepcopy(self._config)
+        warnings.warn(
+            '"get_config" method is deprecated. Use "config" property instead.',
+            DeprecationWarning
+        )
+        return self.config
+
+    @config.setter
+    def config(self, value: MutableMapping):
```

Comment:
I understand that the main advantage of `MappingProxyType` over plain dicts is that we can avoid deepcopying it. But what about if a user provides a dict to the setter and then modifies that dict? It would cause the dict inside aiocache to be modified too right? This might introduce subtle bugs :/

Reply:
that's a valid point, maybe we can still have to use `deepcopy` here